esp/esp/program/models/class_.py

```diff
@@ -699,6 +698,12 @@ def assign_start_time(self, first_event):
         if availability:
             for room in current_rooms:
                 self.assign_room(room)
+
+    def end_time(self):
+        """Returns the meeting time for this section with the latest end time"""
+        all_times = self.meeting_times.order_by("-end")
+        if all_times: return all_times[0]
+        else: return []
```

Comment:
Sections have no meeting_times when they aren't scheduled (for example because they are rejected). If they aren't scheduled, they shouldn't end up on student schedules, so that shouldn't be a problem for the other code in this branch.